### PR TITLE
Replace vitess:base with vitess:lite images for docker-compose services

### PR DESCRIPTION
**Pull request by jawabuu**
This PR replaces the huge `vitess: base` images >2G with the `vitess:lite` image.
The objective is to make it quicker for someone to spin up a docker-compose example.
A secondary objective is to mirror operator design and discard the `InitShardMaster` introduced [here](https://github.com/vitessio/vitess/pull/6584) by implementing orchestrator.
@sougou @shlomi-noach @deepthi

### CODE REVIEW

File: examples/compose/config/init_db.sql [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+# User for Orchestrator (https://github.com/openark/orchestrator).
+# TODO: Reenable when the password is randomly generated.
+CREATE USER 'orc_client_user'@'%' IDENTIFIED BY 'orc_client_user_password';
+GRANT SUPER, PROCESS, REPLICATION SLAVE, RELOAD
+  ON *.* TO 'orc_client_user'@'%';
+GRANT SELECT
+  ON _vt.* TO 'orc_client_user'@'%';
+FLUSH PRIVILEGES;
+RESET SLAVE ALL;
+RESET MASTER;
```
Comment (shlomi-noach) [inline examples/compose/config/init_db.sql:77]: I'm just wondering, what changed that we need this addition of database cleanup, that we got away without thus far?

**Review by shlomi-noach — COMMENTED** [review]
A couple questions inline for clarifications.

File: examples/compose/config/init_db.sql [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  value VARCHAR(255) NOT NULL,
+  db_name VARBINARY(255) NOT NULL,
+  PRIMARY KEY (db_name, name)
+  ) ENGINE=InnoDB;
+CREATE TABLE IF NOT EXISTS _vt.shard_metadata (
+  name VARCHAR(255) NOT NULL,
+  value MEDIUMBLOB NOT NULL,
+  db_name VARBINARY(255) NOT NULL,
+  PRIMARY KEY (db_name, name)
+  ) ENGINE=InnoDB;
```
[the remark this answers is not in the document]
Comment (jawabuu) [reply examples/compose/config/init_db.sql:33]: Using the `vitess:lite` image and following the operator pattern, I have included the init_db.sql as it doesn't seem to be present.
The `vitess:base` image has the init_db.sql hence we had no reason to add it.

File: examples/compose/config/init_db.sql (hunk 1 above)
Comment (jawabuu) [reply examples/compose/config/init_db.sql:77]: I believe this has always been present
https://github.com/vitessio/vitess/blob/master/config/init_db.sql

Comment (jawabuu) [reply examples/compose/config/init_db.sql:77]: Checking the file, it was added here
https://github.com/vitessio/vitess/commit/a9cf065995cf264a24223c4ad8443d6df78abc3e

File: examples/compose/config/init_db.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (shlomi-noach) [reply examples/compose/config/init_db.sql:33]: Got it. Thanks for clarifying!

[the remark this answers is not in the document]
Comment (jawabuu) [reply examples/compose/config/init_db.sql:33]: Hey @shlomi-noach I have completed initial integration of orchestrator. Should I open a new pull request or push to this branch?
https://github.com/jawabuu/vitess/tree/use-vtorc

[the remark this answers is not in the document]
Comment (shlomi-noach) [reply examples/compose/config/init_db.sql:33]: @jawabuu a new pull request -- thanks so much!

[the remark this answers is not in the document]
Comment (jawabuu) [reply examples/compose/config/init_db.sql:33]: Welcome. Much appreciated.

File: examples/compose/config/init_db.sql (hunk 1 above)
Comment (sougou) [reply examples/compose/config/init_db.sql:77]: I was actually meaning to add these statements here in order to deprecate `InitShardMaster`, and found that somebody had already added them.

Without these, you cannot initialize a cluster, and we were relying on `InitShardMaster` to do this. After this change, a simple `TabletExternallyReparented`, or `EmergencyReparentShard` coud work. But then, I got attracted to a new shiny called `vtorc`.

File: examples/compose/schemaload.sh [hunk 3]
```diff
@@ -40,6 +41,8 @@ if [ ! -f schema_run ]; then
   vtctlclient -server vtctld:$GRPC_PORT ApplyVSchema -vschema_file /script/${vschema_file} $KEYSPACE || \
   vtctlclient -server vtctld:$GRPC_PORT ApplyVSchema -vschema "$(cat /script/${vschema_file})" $KEYSPACE
   
+  echo "List All Tablets"
+  vtctlclient -server vtctld:$GRPC_PORT ListAllTablets
```
Comment (sougou) [inline examples/compose/schemaload.sh:45]: We can look at (later) simplifying this to just use vtgate.

**Review by sougou — APPROVED** [review]
Nice work. I didn't verify that this works, but the changes look good. If we encounter new problems, we can always fix forward.

File: examples/compose/config/init_db.sql (hunk 1 above)
Comment (jawabuu) [reply examples/compose/config/init_db.sql:77]: `vtorc` is amazing and it's implemented in https://github.com/vitessio/vitess/pull/7009

**Pull request merged**